core/templates/dev/head/pages/exploration-editor-page/services/exploration-data.service.ts

```diff
@@ -52,12 +55,16 @@ angular.module('oppia').factory('ExplorationDataService', [
     var resolvedAnswersUrlPrefix = (
       '/createhandler/resolved_answers/' + explorationId);
     var explorationDraftAutosaveUrl = '';
-    if (GLOBALS.can_edit) {
-      explorationDraftAutosaveUrl = (
-        '/createhandler/autosave_draft/' + explorationId);
-    } else if (GLOBALS.can_voiceover) {
-      explorationDraftAutosaveUrl = (
-        '/createhandler/autosave_voiceover_draft/' + explorationId);
+
+    if (ContextService.getPageContext() !== PAGE_CONTEXT.COLLECTION_EDITOR) {
+      // TODO (Jamesjay4199): Remove this check after #7221 is fixed.
+      if (GLOBALS.can_edit) {
+        explorationDraftAutosaveUrl = (
+          '/createhandler/autosave_draft/' + explorationId);
+      } else if (GLOBALS.can_voiceover) {
+        explorationDraftAutosaveUrl = (
+          '/createhandler/autosave_voiceover_draft/' + explorationId);
+      }
```

Comment:
Does this belong here?

Reply:
Yes
Removing interaction specs global meant no more global for collection-editor page but this service get's loaded by the collection editor and that causes `GLOBALS is undefined` error